src/main/java/TaskControl.java

```diff
@@ -0,0 +1,101 @@
+public class TaskControl {
+    private final Task[] tasks;
+    private int taskCount;
+
+    public TaskControl() {
+        int MAX_TASKS = 100;
+        this.tasks = new Task[MAX_TASKS];
+        taskCount = 0;
+    }
+
+    public int getTaskCount() {
+        return taskCount;
+    }
+
+    public void printList() {
+        if (taskCount == 0) {
+            System.out.println("I have not been given any tasks! Start by creating an event, todo or deadline :)");
+        } else {
+            System.out.println("Here is your list!");
+            for (int i = 0; i < taskCount; i++) {
+                System.out.println((i + 1) + ". " + tasks[i].toString());
+            }
+        }
+    }
+
+    public void toggleTaskStatus(int taskIndex, String status) throws IndexOutOfBoundsException {
```

Comment:
Might have too much nesting, maybe abstracting and applying SLAP would be better